File: backend/src/main/java/by/fintrack/entity/Currency.java
package by.fintrack.entity;

public enum Currency {
    BYN("BYN", "Br"),
    USD("USD", "$"),
    EUR("EUR", "€"),
    RUB("RUB", "₽");

    public final String code;

    public final String symbol;

    Currency(String code, String symbol) {
        this.code = code;
        this.symbol = symbol;
    }

    public static Currency fromCode(String code) {
        for (Currency c : values()) {
            if (c.code.equalsIgnoreCase(code)) return c;
        }
        return BYN;
    }
}

File: mobile/lib/core/models/currency.dart
enum Currency {
  BYN(code: 'BYN', symbol: 'BYN', decimalDigits: 0),
  USD(code: 'USD', symbol: '\$', decimalDigits: 2),
  EUR(code: 'EUR', symbol: '€', decimalDigits: 2),
  RUB(code: 'RUB', symbol: '₽', decimalDigits: 2);

  final String code;
  final String symbol;
  final int decimalDigits;

  const Currency({
    required this.code,
    required this.symbol,
    required this.decimalDigits,
  });

  factory Currency.fromCode(String? code) {
    if (code == null) return BYN;
    return values.firstWhere(
          (c) => c.code.toUpperCase() == code.toUpperCase(),
      orElse: () => BYN,
    );
  }

  String get displayName => '$code ($symbol)';
}